packages/slope/src/adapter.ts

```diff
@@ -0,0 +1,194 @@
+import {
+    BaseSignerWalletAdapter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletNotConnectedError,
+    WalletDisconnectionError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface SlopeWallet {
+    connect(): Promise<{
+        msg: string;
+        data: {
+            publicKey?: string;
+        };
+    }>;
+    disconnect(): Promise<{
+        msg: string;
+    }>;
+    signTransaction(message: string): Promise<{
+        msg: string;
+        data: {
+            publicKey?: string;
+            signature?: string;
+        }
```

Comment:
Great, I like that you added these as possibly undefined and then check for that.

Reply:
😊